iree/compiler/Dialect/Flow/IR/FlowOpFolders.cpp

```diff
@@ -429,11 +429,74 @@ struct ConvertDispatchInputLoadOfTensorToSubTensor
     return success();
   }
 };
+
+// Pulls in constant index operands as static arguments and return type with the
+// new static sizes.
+struct CanonicalizeStaticOperands
+    : public OpRewritePattern<DispatchTensorLoadOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(DispatchTensorLoadOp loadOp,
+                                PatternRewriter &rewriter) const override {
+    if (loadOp.strides().empty() || loadOp.sizes().empty() ||
```

Comment:
This could be cleaned up a bit if you could have a builder method for `dispatch.tensor.load` that takes `ArrayRef<OpFoldResult>` for offsets, sizes and strides.